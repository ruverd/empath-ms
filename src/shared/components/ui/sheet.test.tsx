import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import {
  Sheet,
  SheetContent,
  SheetDescription,
  SheetHeader,
  SheetTitle,
  SheetTrigger,
} from "./sheet";

describe("Sheet", () => {
  it("should render sheet trigger button", () => {
    render(
      <Sheet>
        <SheetTrigger>Open Sheet</SheetTrigger>
        <SheetContent>Content</SheetContent>
      </Sheet>
    );
    expect(screen.getByText("Open Sheet")).toBeInTheDocument();
  });

  it("should render sheet content when triggered", () => {
    render(
      <Sheet>
        <SheetTrigger>Open Sheet</SheetTrigger>
        <SheetContent>Sheet Content</SheetContent>
      </Sheet>
    );

    fireEvent.click(screen.getByText("Open Sheet"));
    expect(screen.getByText("Sheet Content")).toBeInTheDocument();
  });

  it("should render sheet with header components", () => {
    render(
      <Sheet>
        <SheetTrigger>Open Sheet</SheetTrigger>
        <SheetContent>
          <SheetHeader>
            <SheetTitle>Test Title</SheetTitle>
            <SheetDescription>Test Description</SheetDescription>
          </SheetHeader>
        </SheetContent>
      </Sheet>
    );

    fireEvent.click(screen.getByText("Open Sheet"));
    expect(screen.getByText("Test Title")).toBeInTheDocument();
    expect(screen.getByText("Test Description")).toBeInTheDocument();
  });

  it("closes sheet when close button is clicked", () => {
    render(
      <Sheet>
        <SheetTrigger>Open Sheet</SheetTrigger>
        <SheetContent>Sheet Content</SheetContent>
      </Sheet>
    );

    fireEvent.click(screen.getByText("Open Sheet"));
    const closeButton = screen.getByRole("button", { name: /close/i });
    fireEvent.click(closeButton);

    // The content should be removed from the document
    expect(screen.queryByText("Sheet Content")).not.toBeInTheDocument();
  });

  it("should match snapshot with all subcomponents", () => {
    const { container } = render(
      <Sheet>
        <SheetTrigger>Open Sheet</SheetTrigger>
        <SheetContent>
          <SheetHeader>
            <SheetTitle>Test Title</SheetTitle>
            <SheetDescription>Test Description</SheetDescription>
          </SheetHeader>
          Sheet Content
        </SheetContent>
      </Sheet>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
