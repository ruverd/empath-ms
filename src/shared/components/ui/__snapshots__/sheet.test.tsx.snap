// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sheet > should match snapshot with all subcomponents 1`] = `
<button
  aria-controls="radix-:rc:"
  aria-expanded="false"
  aria-haspopup="dialog"
  data-state="closed"
  type="button"
>
  Open Sheet
</button>
`;
